mono/metadata/sgen-mono.c

```diff
@@ -2217,14 +2217,127 @@ sgen_client_scan_thread_data (void *start_nursery, void *end_nursery, gboolean p
 		   will raise a STATUS_GUARD_PAGE_VIOLATION and the process will crash. This code uses
 		   VirtualQuery() to determine whether stack_start points into the guard page and then
 		   updates aligned_stack_start to point at the next non-guard page. */
+		//
```

Comment:
When we pick solution, rewrite and reduce comment to correctly match selected solution.

Reply:
Yeah, it was a long comment, and almost correct. It was easier to amend than to edit. I can try.

Reply:
The previous long comment was duplicated twice.
New comment is longer. Put in one place and reference from other?
For now I put it in guard-page.txt.